pkg/controllers/deprovisioning/controller.go

```diff
@@ -158,7 +158,7 @@ func (c *Controller) deprovision(ctx context.Context, deprovisioner Deprovisione
 	if err != nil {
 		return false, fmt.Errorf("computing deprovisioning decision, %w", err)
 	}
-	if cmd.action == actionDoNothing {
+	if len(cmd.candidates) == 0 {
```

Comment:
I'd feel more comfortable checking that a `Command` is completely empty over checking one part of the field

Reply:
Synced offline, going with this approach since you can't compare structs that have pointer values without a deepcopy.